search.go

```diff
@@ -105,26 +111,49 @@ type FacetRequest struct {
 }
 
 func (fr *FacetRequest) Validate() error {
-	if len(fr.NumericRanges) > 0 && len(fr.DateTimeRanges) > 0 {
+	nrCount := len(fr.NumericRanges)
+	drCount := len(fr.DateTimeRanges)
+	if nrCount > 0 && drCount > 0 {
 		return fmt.Errorf("facet can only conain numeric ranges or date ranges, not both")
 	}
-
-	nrNames := map[string]interface{}{}
-	for _, nr := range fr.NumericRanges {
-		if _, ok := nrNames[nr.Name]; ok {
-			return fmt.Errorf("numeric ranges contains duplicate name '%s'", nr.Name)
+	rangeSet := false
+	if nrCount > 0 {
+		nrNames := map[string]interface{}{}
+		for _, nr := range fr.NumericRanges {
+			if _, ok := nrNames[nr.Name]; ok {
+				return fmt.Errorf("numeric ranges contains duplicate name '%s'", nr.Name)
+			}
+			nrNames[nr.Name] = struct{}{}
+			if !rangeSet && (nr.Min != nil || nr.Max != nil) {
+				rangeSet = true
+			}
+		}
+		if !rangeSet {
+			return fmt.Errorf("numeric range query must specify either min, max or both")
```

Comment:
Oh I see, you mean to say that 
```
 "numeric_ranges": [
        {
          "name": "high"
        },
        {
          "name": "low",
          "min": 7
        }
      ]
``` .  
is an invalid query  ?  [I was thinking in the whole numeric_range [] array we have to specify the boundaries for any of the entries]